File: src/audio_stegano_app/lib/core/stego/stego_embed_mode.dart
/// Embedding path — `train/embed_message.m` `embed_mode`.
enum StegoEmbedMode {
  /// `xor_only`: message bits XOR logistic key, then LSB.
  xorOnly,

  /// `ae_xor`: 8-bit blocks through trained autoencoder, then XOR, then LSB.
  aeXor,
}

extension StegoEmbedModeJson on StegoEmbedMode {
  static StegoEmbedMode fromConfig(String? raw) {
    switch ((raw ?? '').trim().toLowerCase()) {
      case 'ae_xor':
      case 'aexor':
      case 'ae':
        return StegoEmbedMode.aeXor;
      case 'xor_only':
      case 'xoronly':
      case 'xor':
      default:
        return StegoEmbedMode.xorOnly;
    }
  }

  String get configValue => switch (this) {
        StegoEmbedMode.xorOnly => 'xor_only',
        StegoEmbedMode.aeXor => 'ae_xor',
      };
}

File: src/audio_stegano_desktop/src/AudioStegano.Core/Stego/StegoEmbedMode.cs
namespace AudioStegano.Core.Stego;

/// <summary><c>train/embed_message.m</c> <c>embed_mode</c>.</summary>
public enum StegoEmbedMode
{
    XorOnly,
    AeXor,
}

public static class StegoEmbedModeParser
{
    public static StegoEmbedMode FromConfig(string? raw) => (raw ?? "").Trim().ToLowerInvariant() switch
    {
        "ae_xor" or "aexor" or "ae" => StegoEmbedMode.AeXor,
        _ => StegoEmbedMode.XorOnly,
    };

    public static string ToConfigValue(this StegoEmbedMode mode) => mode switch
    {
        StegoEmbedMode.AeXor => "ae_xor",
        _ => "xor_only",
    };
}
